server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -279,4 +283,17 @@ export class AssetRepository implements IAssetRepository {
       relations: ['exifInfo'],
     });
   }
+
+  async getExistingAssets(userId: string, checkDuplicateAssetDto: CheckExistingAssetsDto): Promise<CheckExistingAssetsResponseDto> {
+    const existingAssets = await this.assetRepository.find({
+      select: {deviceAssetId: true},
+      where: {
+        deviceAssetId: In(checkDuplicateAssetDto.deviceAssetIds),
+        deviceId: checkDuplicateAssetDto.deviceId,
```

Comment:
Should we include deviceId in the check here? The rest of the duplicate detection doesn't AFAIK.

Reply:
These two are the only clues you have from the client, so I think they must be present to distinguish which asset from which device.